scala/private/phases/phase_coverage_runfiles.bzl

```diff
@@ -14,7 +14,7 @@ def phase_coverage_runfiles(ctx, p):
     if ctx.configuration.coverage_enabled and _coverage_replacements_provider.is_enabled(ctx):
         coverage_replacements = _coverage_replacements_provider.from_ctx(
             ctx,
-            base = p.coverage.replacements,
+            base = p.coverage.replacements if ctx.coverage_instrumented() else {},
```

Comment:
I think I'm missing something- this is the only prod code change, right?
If so can you explain how it solves the test targets?

Reply:
It means - if current rule is filtered out don't include its sources. `--instrument_test_targets` controls if by default test target sources should be filtered out.

Reply:
Thanks! I missed the original issue.
I thought we never instrumented and that you added that under the flag.
Now I understand we always instrumented and now you use the flag to toggle it.
Thanks!